components/tag-list/tag-list.js

```diff
@@ -39,41 +63,157 @@ class TagList extends ArrowKeysMixin(LitElement) {
 		/** @ignore */
 		this.arrowKeysDirection = 'leftrightupdown';
 		this._items = [];
+		this._resizeObserver = null;
+		this._showHiddenTags = false;
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._resizeObserver) this._resizeObserver.disconnect();
 	}
 
 	firstUpdated(changedProperties) {
 		super.firstUpdated(changedProperties);
 
-		requestAnimationFrame(() => {
-			this._items = this._getTagListItems();
-			this._items.forEach((item, index) => {
-				item.setAttribute('tabIndex', index === 0 ? 0 : -1);
-			});
-		});
+		this._container = this.shadowRoot.querySelector('.tag-list-outer-container');
+		this._resizeObserver = new ResizeObserver((e) => requestAnimationFrame(() => this._handleResize(e)));
+		this._resizeObserver.observe(this._container);
 	}
 
 	render() {
+		let hiddenCount = 0;
+		let hasHiddenTags = false;
+		this._items.forEach((element, index) => {
+			if (index >= this._chompIndex) hasHiddenTags = true;
+			if (!this._showHiddenTags && index >= this._chompIndex) {
+				hiddenCount++;
+				element.setAttribute('data-is-chomped', '');
+			} else {
+				element.removeAttribute('data-is-chomped');
+			}
+		});
+
+		let button = null;
+		if (hasHiddenTags) {
+			button = this._showHiddenTags ? html`
+				<d2l-button-subtle slim text="Show Less" @click="${this._toggleHiddenTagVisibility}" class="d2l-tag-list-button"></d2l-button-subtle>
+			` : html`
+				<d2l-button-subtle slim text="+ ${hiddenCount} more" @click="${this._toggleHiddenTagVisibility}" class="d2l-tag-list-button"></d2l-button-subtle>
+			`;
+		}
+
 		const list = html`
 			<div role="list" class="tag-list-container" aria-describedby="d2l-tag-list-description">
-				<slot></slot>
+				<slot @slotchange="${this._handleSlotChange}"></slot>
+				${button}
 			</div>
 		`;
+
+		const outerContainerStyles = {
+			maxHeight: this._showHiddenTags ? 'unset' : `${(this._itemHeight + MARGIN_TOP_HEIGHT) * this._lines}px`
```

Comment:
When we click the button to show the hidden tags, we want it to take up all the room, else we want it limited based on number of lines and line height